src/order.jl

```diff
@@ -219,15 +219,28 @@ function vertices(
     g::BipartiteGraph, ::Val{side}, ::DynamicDegreeBasedOrder{degtype,direction}
 ) where {side,degtype,direction}
     other_side = 3 - side
+    # compute dist-2 degrees in an optimized way
+    n = nb_vertices(g, Val(side))
+    degrees_dist2 = zeros(Int, n)
+    dist2_neighbor = falses(n)
+    for v in vertices(g, Val(side))
+        fill!(dist2_neighbor, false)
+        for w1 in neighbors(g, Val(side), v)
+            for w2 in neighbors(g, Val(other_side), w1)
+                dist2_neighbor[w2] = true
+            end
+        end
+        degrees_dist2[v] = sum(dist2_neighbor)
```

Comment:
I suggest to directly update `degrees_dist2[v]` after line 230 with a `degrees_dist2[v] += 1`.

Reply:
Update: I am wrong if we can visit `w2` multiple times (`dist2_neighbor[w2] = true`).
If it's the case, forget my comment.

Reply:
yes we can visit multiple times